### ISSUE TITLE
Error with a decorator with Ember 3.5

### ISSUE THREAD

**Issue by Bartheleway** [alexlafroscia/ember-steps#128]
Hi,

I don't know what cause this weird error but I get a : `Assertion Failed: The @action decorator must be applied to functions` from ember-steps.

It looks like it comes from the fact I'm using Ember 3.5 (and all the stuff that comes with it). I've seen that you're working on a compatibility fix, any ways that you publish a beta ? That way I will be happy to test it and tell you if it fix the situation.

**Comment by lougreenwood** [comment]
I just came to also report this issue too.

Seems to be related to using ember-decorators 3.1 & stage2 decorators... `ember-steps` is currently using ED 3.0, but if a later add-on in the dependency tree requires 3.1 this will be used instead.

I triggered this error quite simply by forcing ember-steps to use `ember-decorators` 3.1 in package.json (because I was getting conflicts when I upgraded my app to use ED 3.1):

```
    "resolutions": {
        "ember-steps/**/@ember-decorators/component": "^3.1.2",
        "ember-steps/**/@ember-decorators/object": "^3.1.2"
    },
```

**Comment by lougreenwood** [comment]
For me, it looks like the following decorator triggers the error - I'm wondering if this might be an issue in Ember Decorators, so I've also asked in `#e-decorators` on Discord.

https://github.com/alexlafroscia/ember-steps/blob/master/addon/components/step-manager.ts#L123

<img width="728" alt="screenshot 2018-12-01 at 13 48 10" src="https://user-images.githubusercontent.com/12051736/49328909-d2b43300-f56f-11e8-868d-e4330fd6cfe1.png">

<img width="1298" alt="screenshot 2018-12-01 at 13 49 18" src="https://user-images.githubusercontent.com/12051736/49328914-e95a8a00-f56f-11e8-98fd-118340f65084.png">

**Comment by lougreenwood** [comment]
Seems this is a bug in `ember-decorators`, fixed by @pzuraq here - https://github.com/ember-decorators/ember-decorators/pull/324

**Comment by Bartheleway** [comment]
I confirm the bug is gone 👍

**Comment by Bartheleway** [comment]
As writen in `package.json`, ember-steps depend on `ember-decorators : "^3.0.1"` so now that they fix the bug it will upgrade by itself. You just have to run yarn again on your project (that's what I did).

**Comment by pzuraq** [comment]
This is true, but you will still be getting a warning message about updating to ember-cli-typescript v2, which I think is what @lougreenwood is referring to

**Comment by alexlafroscia** [comment]
Sorry about this issues! I’ll do those updates soon.

**Comment by pzuraq** [comment]
Not a rush, that’s why we ended up making it a warning so folks could update incrementally. Sorry about the bug that caused the main issues in the first place!

**Comment by alexlafroscia** [comment]
Taking care of this right now.

**Comment by pzuraq** [comment]
I have not, this seems to be coming from the transforms somehow though. `isClassPrivateMethod` must be referring to a check for actual private methods, which didn't exist in babel 6.

Have you bumped `@ember-decorators/babel-transforms` to v3.1.0?

**Comment by alexlafroscia** [comment]
I get a warning when upgrading them about needing to upgrade `ember-cli-typescript` to one that plays nicely with Stage 2 decorators.

If I upgrade `ember-cli-typescript` first, I get an error about not having legacy decorators support. As soon as I install `@ember-decorators/babel-transforms` I get the error above about `isClassPrivateMethod` not being a function.

**Comment by alexlafroscia** [comment]
> Have you bumped @ember-decorators/babel-transforms to v3.1.0?

Somehow, this add-on never had it as a dependency. Installing either latest version doesn't seem to help at all, though.

**Comment by alexlafroscia** [comment]
The PR above has what I've tried... I don't know why that isn't working though

**Comment by lougreenwood** [comment]
@alexlafroscia & @alexlafroscia 

I wonder if this is related to Babel 7.2 which was just released? It seems that Babel 7.2 includes stage 3 private class methods: https://github.com/babel/babel/commit/0859535b620ddbc0a835c7882efbf3b5660df3b0

The reason I noticed this is because an application which I just attempted to update to ED 3.1.4 also got the same error...

**Comment by alexlafroscia** [comment]
Yeah, I just saw this tweet

https://twitter.com/NicoloRibaudo/status/1069977807437119488

Will try again today!
